feat: replace ServerAt by the Server it's pointing to

ca-d opened the pull request:
- **test(hash): ServerAt reference transparency**
- **feat(hash): ServerAt reference transparency**

ca-d committed:
test(hash): ServerAt reference transparency

ca-d committed:
feat(hash): ServerAt reference transparency

ca-d committed:
test(hash): transitive transparency of ServerAt references

danyill commented on hash.ts:
```diff
@@ -10,6 +10,54 @@ export type Description = Record<string, string | string[]> & {
 };
 
+function findAccessPointReferences(e: Element): Element[] {
+  let serverAt = e.querySelector(':scope>ServerAt, :scope>Server');
```
Could we call this `server`

ca-d replied:
Agreed.

danyill commented on hash.ts:
```diff
@@ -922,9 +982,21 @@ export function hasher(
 
+  function describeAccessPoint(e: Element) {
+    const description = {
+      ...describeAttributes(e),
```
We must have a `describeChildren` here

ca-d replied:
Done.

danyill commented on hash.ts:
```diff
@@ -10,6 +10,54 @@ export type Description = Record<string, string | string[]> & {
+  let serverAt = e.querySelector(':scope>ServerAt, :scope>Server');
+  let apName = serverAt?.getAttribute('apName');
+  while (serverAt && serverAt.tagName !== 'Server') {
+    serverAt =
+      serverAt
+        .closest('IED')
+        ?.querySelector(
+          `:scope>AccessPoint[name="${apName}"]>Server, :scope>AccessPoint[name="${apName}"]>ServerAt`,
+        ) ?? null;
```
There is an issue with selectors like this not respecting filter criteria.

ca-d replied:
Done.

ca-d committed:
refactor(hash): rename serverAt -> server

ca-d committed:
fix: also hash AccessPoint's non-server children

ca-d committed:
fix: respect filter criteria when dereferencing ServerAt

ca-d commented:
@danyill Thank you for the valuable feedback! I've addressed the issues you raised and would be glad if you would take another look at the current state of the code.

danyill reviewed (approved):
I think, insofar as I understand it 😜 it looks good to me, thank you for the opportunity to study this 🙇‍♂️
